src/utils/externalLinkHelper.js

```diff
@@ -0,0 +1,36 @@
+import { useState } from 'react';
+
+const useExternalLinkState = () => {
```

Comment:
🔧 add `dispatch` props here (it's used in multiple places) and import `triggerCodeRun` in this file and remove the prop entirely